sinatra.gemspec

```diff
@@ -46,7 +46,9 @@ RubyGems 2.0 or newer is required to protect against public gem pushes. You can
   s.required_ruby_version = '>= 2.6.0'
 
   s.add_dependency 'mustermann', '~> 3.0'
-  s.add_dependency 'rack', '~> 2.2', '>= 2.2.4'
+  s.add_dependency 'rack', '>= 3.0.0.beta1', '< 4'
+  s.add_dependency 'rackup', '>= 2.0.0', '< 3'
+  s.add_dependency 'rack-session', '>= 2.0.0', '< 3'
```

Comment:
`rack-session` is required only in `rack-protection/lib/rack/protection.rb`. maybe removing this line and add to `rack-protection/rack-protection.gemspec`?
```suggestion
```

Reply:
Thanks for catching this, not having the `rack-session` dependency in `rack-protection` actually makes it broken. It isn't discovered in the specs due to this line https://github.com/sinatra/sinatra/blob/d66fa62d05de2d10de9bf8d866c054a425c85885/rack-protection/Gemfile#L14

Reply:
I will note though, that solving https://github.com/sinatra/sinatra/issues/1945 means adding `rack-session` as a dependency to Sinatra. I think we would like to address that issue in Sinatra 4.0 too.

Reply:
Tests adjusted in main (dcdebe981074f0a79a0d8384d149c3d2db831a76) and I've moved the dependency to rack-protection now.